recovery.md

```diff
@@ -0,0 +1,82 @@
+# The recovery partition
+
+On Ubuntu Core 20 systems there is a new system-recovery partition. It
+is big enough to contain a certain number of recovery "systems" that
+can then be used to recover or reinstall a broken system.
+
+A full "seed" to reinstall on a generic amd64 system is roughly 280MB
+so the recovery partition is sized to 600MB to be able to store two
+recovery systems by default.
+
+The system-recovery partition is next to the system-boot
+partition. Both are vfat as required by UEFI.
+
+On full disk encrypted systems the system-boot partition contains the
+unpacked kernel(s) to boot init initramfs that then unencrypts the
+"writable" partition. The partitions are separate because we want to
+write to the recovery partition as rarely as possible to avoid
+filesystem issues.
+
+# Recovery system file layout
+
+The revery partition contains the following file layout:
+
+/system/<name>/snaps/{base.snap,kernel.snap,other...}
+/system/<name>/assertions.txt
```

Comment:
not a fan of the ".txt" extetnsion here, also unclear why we shouldn't support multiple streams

Reply:
I think this should be a assertions.d style directory.